### PR TITLE
Data processing with models

### PR DESCRIPTION
**Pull request by why-a-en**
### Branch changes

- Created models
- Used model to process data
- Added custom methods

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  after_create :post_counter
+  after_destroy :post_counter
+
+  def recent_comments
+    comments.last(5).reverse
+  end
+
+  def post_counter
+    author.update(posts_counter: author.posts.count)
+  end
```
Comment (DeliceLydia) [inline app/models/post.rb:17]: ## Optional
- [ ] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `post_counter` method every time a new post has been saved to the database, then make the `post_counter` method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_create :likes_counter
+  after_destroy :likes_counter
+
+  def likes_counter
```
Comment (DeliceLydia) [inline app/models/like.rb:10]: ## Optional
- [ ] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `likes_counter` method every time a new like has been saved to the database, then make the `likes_counter` method private

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_create :comments_counter
+  after_destroy :comments_counter
+
+  def comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (DeliceLydia) [inline app/models/comment.rb:12]: ## Optional
- [x] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `comments_counter` method every time a new comment has been saved to the database, then make the `comments_counter` method private

**Review by DeliceLydia — APPROVED** [review]
## Hi @anbehindY, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- The associations between models are set up correctly
- No linter errors



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/comment.rb (hunk 3 above)
Comment (why-a-en) [reply app/models/comment.rb:12]: Thank you @DeliceLydia

**Pull request merged**